src/services/dashboards/widgets/_components/type.ts

```diff
@@ -0,0 +1,42 @@
+import type { Currency } from '@/store/modules/display/config';
+import type { CurrencyRates } from '@/store/modules/display/type';
+
+interface TextOption {
+    type: 'text';
+    align?: 'left'|'right';
+    icon?: string;
+}
+
+interface CostOption {
+    type: 'cost';
+    currencyLabel: string;
+    currency: Currency;
+    currencyRates: CurrencyRates;
+}
+interface PercentOption {
+    type: 'percent';
+}
+interface SizeOption {
+    type: 'size';
+    display_unit?: 'BYTES | KB | MB | GB | TB | PB';
+    source_unit?: 'BYTES | KB | MB | GB | TB | PB';
+}
+
+export interface Field {
+    name: string;
+    width?: string;
+    label?: string;
+    textAlign?: 'left'|'right'|'center';
+    styleOptions: TextOption;
+    dataOptions: CostOption | PercentOption | SizeOption;
+    detailOptions?: {
+        enabled: boolean;
+        type: 'modal'|'popover';
+    }
+    tooltipText?: string;
+}
```

Comment:
suggestion: Is it final version? If it is, I think some properties must be updated.

I think badge case must be added.
Also, what about remove textAlign, width properties that are not necessary for widget data table?

Reply:
It's not the final version, but I have a question. 
1. I removed the badge because it was designated as below, but would it be better to add it?
    ![image](https://user-images.githubusercontent.com/50662170/204428265-62477eaa-142d-4c01-8c16-8ee1e58b2609.png) 
2. textAlign is added in case the header(th) and data(td) are misaligned. I checked the design again, it's okay to remove it.
3. I added the width just in case, should I remove it?

Reply:
1. hm... I think it would be better to ask to designers!
2. I understand. Then, the name of the prop is really confused with `styleOptions.align` property. What about to add `headerOptions` or somethin for column headers?
3. If you are gonna implement width, it would be alright. But without implementation, developers can be confused I think...